### PR TITLE
Fixed Java Core 894 - Replication.docIds API discrepancy

### PR DESCRIPTION
**Pull request by hideki**
- ported codes from couchbase/couchbase-lite-ios@551ce20

### CODE REVIEW

File: src/main/java/com/couchbase/lite/replicator/Replication.java [hunk 1]
```diff
@@ -93,7 +94,42 @@
     /**
      * Properties of the replicator that are saved across restarts
      */
-    protected Map<ReplicationField, Object> properties;
+    private Map<ReplicationField, Object> properties;
+
+    /**
+     * Currently only used for test
+     */
+    public Map<String, Object> getProperties() {
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/Replication.java]: As this only for testing, I think you can use package-private (no explicit modifier) for this method.

File: src/main/java/com/couchbase/lite/replicator/Replication.java [hunk 2]
```diff
@@ -93,7 +94,42 @@
...
+            props.put("target", db.getName());
+        }
+        else{
+            props.put("source", db.getName());
+            props.put("target", remote);
+        }
+
+        // TODO: customProperties are not implimented for CBL Android/Java
+        // if (_customProperties)
+        //  [props addEntriesFromDictionary: _customProperties];
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/Replication.java]: Please remove this TODO as it's not applicable for Java/Android.

File: src/main/java/com/couchbase/lite/replicator/Replication.java [hunk 3]
```diff
@@ -364,7 +400,8 @@ public ReplicationStatus getStatus() {
             return ReplicationStatus.REPLICATION_OFFLINE;
         } else if (replicationInternal.stateMachine.isInState(ReplicationState.IDLE)) {
             return ReplicationStatus.REPLICATION_IDLE;
-        } else if (replicationInternal.stateMachine.isInState(ReplicationState.STOPPED)) {
+        } else if (replicationInternal.stateMachine.isInState(ReplicationState.INITIAL) ||
+                replicationInternal.stateMachine.isInState(ReplicationState.STOPPED)) {
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/Replication.java]: Is this also related to docIDs fixing? or Is this another bug fix?

Comment (hideki) [inline src/main/java/com/couchbase/lite/replicator/Replication.java]: This issue was found during writing replicator state test line. Please see following link.
https://github.com/couchbase/couchbase-lite-android/blob/feature/issue_894_docIDs/src/androidTest/java/com/couchbase/lite/replicator/ReplicationTest.java#L124

**Comment by hideki** [comment]
@pasin PR was updated. Please review this again.
